diesel_codegen_new/src/ast_builder.rs

```diff
@@ -0,0 +1,16 @@
+use syn::{Ident, Ty, Path, PathSegment};
+
+pub fn ty_ident(ident: Ident) -> Ty {
```

Comment:
I think it makes sense for syn to only provide the AST definitions and parsing, and having a separate crate for the AST builder. I definitely think that abstraction needs to exist. I'm not confident enough in the usage patterns that have emerged to really find the right abstractions yet though.